src/huggingface_hub/inference/_generated/_async_client.py

```diff
@@ -1251,6 +1252,55 @@ async def text_to_speech(self, text: str, *, model: Optional[str] = None) -> byt
         """
         return await self.post(json={"inputs": text}, model=model, task="text-to-speech")
 
+    async def token_classification(self, text: str, *, model: Optional[str] = None) -> List[TokenClassificationOutput]:
+        """
+        Perform token classification on the given text.
+        Usually used for sentence parsing, either grammatical, or Named Entity Recognition (NER) to understand keywords contained within text.
+
+        Args:
+            text (`str`):
+                A string to be classified.
+            model (`str`, *optional*):
+                The model to use for the token classification task. Can be a model ID hosted on the Hugging Face Hub or a URL to
+                a deployed Inference Endpoint. If not provided, the default recommended token classification model will be used.
+                Defaults to None.
+
+        Returns:
+            `List[Dict]`: List of token classification outputs containing the entity group, confidence score, word, start and end index.
+
+        Raises:
+            [`InferenceTimeoutError`]:
+                If the model is unavailable or the request times out.
+            `aiohttp.ClientResponseError`:
+                If the request fails with an HTTP error status code other than HTTP 503.
+
+        Example:
+        ```py
+        # Must be run in an async context
+        >>> from huggingface_hub import AsyncInferenceClient
+        >>> client = AsyncInferenceClient()
+        >>> output = await client.token_classification("My name is Sarah Jessica Parker but you can call me Jessica")
+        >>> output
```

Comment:
```suggestion
        >>> await client.token_classification("My name is Sarah Jessica Parker but you can call me Jessica")
```

(same as sync version)